projects/storefrontlib/shared/services/import-export/file-download.service.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@angular/core';
+import { ExportFileOptions } from '../../models/file';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class FileDownloadService {
+  download(fileContent: string, fileOptions: ExportFileOptions): void {
```

Comment:
I think this service should be a generic util for downloading a file, regardless whether it's created from a string, or whether it's an URL from the Web.

The most basic usage I can imagine is:
```typescript
/**
  * Triggers the browser downloading of the file from the given URL.
  * The optional parameter `filename`  allows for defining the suggested filename
  * (including the extension) when saving the file.
**/
download({ url: string, fileName?: string})
```

But our case is a bit more custom:
- we **create** a file from a **string** content
- we decide it's a **text** file
- we decide on the **charset** of the file (add download)
- we decide on the **mime type** of the file
- finally we **store the file content encoded in the URL**

I feel we should have a separate method or service for that sort of logic.
What do you think?